tests/models/test_openrouter.py

```diff
@@ -409,6 +411,70 @@ class FindEducationContentFilters(BaseModel):
     )
 
 
+async def test_openrouter_image_generation(allow_model_requests: None, openrouter_api_key: str) -> None:
+    provider = OpenRouterProvider(api_key=openrouter_api_key)
+    model = OpenRouterModel(
+        model_name='google/gemini-2.5-flash-image-preview',
+        provider=provider,
+    )
+    agent = Agent(model=model, output_type=str | BinaryImage)
```

Comment:
We should also test a scenario with `ImageGenerationTool` but no `output_type=BinaryImage`, where we read the files off of `response.files`, like here:

https://github.com/pydantic/pydantic-ai/blob/d2b08ad12e103e81bd0a9dc0eb7f2a7c3472931d/tests/models/test_openai_responses.py#L6026-L6040